src/System.Net.WebHeaderCollection/src/System/Net/HeaderInfoTable.cs

```diff
@@ -3,88 +3,146 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections;
+using System.Collections.Generic;
 using System.Collections.Specialized;
 
 namespace System.Net
 {
     internal class HeaderInfoTable
     {
-        private static Hashtable HeaderHashTable;
-        private static HeaderInfo UnknownHeaderInfo = new HeaderInfo(string.Empty, false, false, false);
+        private static readonly Func<string, string[]> s_singleParser = value => new[] { value };
+        private static readonly Func<string, string[]> s_multiParser = value => ParseValueHelper(value, isSetCookie: false);
+        private static readonly Func<string, string[]> s_setCookieParser = value => ParseValueHelper(value, isSetCookie: true);
+        private static readonly HeaderInfo s_unknownHeaderInfo = new HeaderInfo(string.Empty, false, false, false, s_singleParser);
+        private static readonly Hashtable s_headerHashTable;
+
+        private static string[] ParseValueHelper(string value, bool isSetCookie)
+        {
+            List<string> tempStringCollection = new List<string>();
+
+            bool inquote = false;
+            int chIndex = 0;
+            char[] vp = new char[value.Length];
```

Comment:
I don't know how important performance is for usage of ParseValueHelper, but if it's important and if we expect value will be relatively short (e.g. frequently < 256 chars), it'd be nice to optimize for such cases by using stack space when it's sufficiently short rather than always allocating a char[].

Reply:
From Google's SPDY research project whitepaper:

> Uncompressed request and response headers. Request headers today vary in size from ~200 bytes to over 2KB. As applications use more cookies and user agents expand features, typical header sizes of 700-800 bytes is common.

Which is around 350-400 chars.

I will re-write this helper function with stack space for performance reason.